### PR TITLE
Update mcse_sd calculation to not use normality assumption.

### LINKED ISSUE #2005: Better mcse_sd
ArviZ is currently using the same mcse_sd that posterior was using, which assumes normality of the sample. The posterior issue
https://github.com/stan-dev/posterior/issues/232 illustrates the problem and the better behavior of new mcse_sd. The posterior PR https://github.com/stan-dev/posterior/pull/233 shows the new code in R, which should be easy to translate to Python.

### PR DESCRIPTION
**Pull request by ahartikainen**
Fixes #2005 

See https://github.com/stan-dev/posterior/pull/233/files

## Checklist

- [ ] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] Includes a sample plot to visually illustrate the changes (only for plot-related functions)
- [ ] New features are properly documented (with an example if appropriate)?
- [ ] Includes new or updated tests to cover the new feature
- [ ] Code style  correct (follows pylint and black guidelines)
- [ ] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

File: arviz/stats/diagnostics.py [hunk 1]
```diff
@@ -826,13 +826,15 @@ def _mcse_sd(ary):
...
         return np.nan
-    ess = _ess_sd(ary)
+    ary = ary - ary.mean()
+    ess = _ess_mean(abs(ary))
+    evar = (ary**2).mean()
+    varvar = ((ary**4).mean() - evar**2) / ess
+    varsd = varvar / evar / 4
     if _numba_flag:
-        sd = np.float(_sqrt(svar(np.ravel(ary), ddof=1), np.zeros(1)))
+        mcse_sd_value = np.float(_sqrt(np.ravel(varsd), ddof=1), np.zeros(1))
```
Comment (aloctavodia) [inline arviz/stats/diagnostics.py:835]:
```suggestion
        mcse_sd_value = np.float(_sqrt(np.ravel(varsd), np.zeros(1)))
```

Comment (aloctavodia) [reply arviz/stats/diagnostics.py:835]: Or this should be the same as before

`np.float(_sqrt(svar(np.ravel(varsd), ddof=1), np.zeros(1)))`

File: arviz/stats/diagnostics.py [hunk 2]
```diff
@@ -826,13 +826,15 @@ def _mcse_sd(ary):
...
     else:
-        sd = np.std(ary, ddof=1)
-    fac_mcse_sd = np.sqrt(np.exp(1) * (1 - 1 / ess) ** (ess - 1) - 1)
-    mcse_sd_value = sd * fac_mcse_sd
+        mcse_sd_value = np.std(varsd, ddof=1)
```
Comment (aloctavodia) [inline arviz/stats/diagnostics.py:837]:
```suggestion
        mcse_sd_value = np.sqrt(varsd)
```

**Review by aloctavodia — APPROVED** [review]

File: arviz/stats/diagnostics.py [hunk 3]
```diff
@@ -732,8 +732,8 @@ def _ess_sd(ary, relative=False):
     ary = np.asarray(ary)
     if _not_valid(ary, shape_kwargs=dict(min_draws=4, min_chains=1)):
         return np.nan
-    ary = _split_chains(ary)
-    return min(_ess(ary, relative=relative), _ess(ary**2, relative=relative))
+    ary = abs(ary - ary.mean())
```
Comment (sethaxen) [inline arviz/stats/diagnostics.py:735]: Based on benchmarks, posterior just switched to this form (see https://github.com/stan-dev/posterior/pull/266):
```suggestion
    ary = (ary - ary.mean())**2
```

**Review by sethaxen — CHANGES_REQUESTED** [review]

File: arviz/stats/diagnostics.py (hunk 3 above)
Comment (ahartikainen) [reply arviz/stats/diagnostics.py:735]: Thanks, I will add this. Need to go through loo to make sure we have the latest changes

Comment (OriolAbril) [reply arviz/stats/diagnostics.py]: By fixing the tests I realized that change should have been done to mcse_sd not to ess_sd. With the changes I just pushed we are back to 100% match with posterior

**Pull request merged**